enhancements/authentication/login-logout-events.md

```diff
@@ -42,67 +43,21 @@ This approach has a couple of drawbacks:
-2. Provide a fully featured audit subsystem in oauth-server compliant with API server policy framework.
-3. Rate limiting of logging login/logout events.
+1. Provide a fully featured audit subsystem in oauth-server compliant with API server policy framework.
+2. Rate limiting of logging login/logout events.
 
 ## Proposal
 
-We propose to add an `audit` struct to `oauth.config.openshift.io/v1` with a `profile` field which configures audit logging granularity.
-
-```go
-// OAuthSpec contains desired cluster auth configuration
-type OAuthSpec struct {
-	// identityProviders is an ordered list of ways for a user to identify themselves.
-	// When this list is empty, no identities are provisioned for users.
-	// +optional
-	// +listType=atomic
-	IdentityProviders []IdentityProvider `json:"identityProviders,omitempty"`
-
-	// tokenConfig contains options for authorization and access tokens
-	TokenConfig TokenConfig `json:"tokenConfig"`
-
-	// templates allow you to customize pages like the login page.
-	// +optional
-	Templates OAuthTemplates `json:"templates"`
-	// audit specifies what should be audited in the context of OAuthServer.
-	// +optional
-	// +kubebuilder:default:={"profile":"WriteLoginEvents"}
-	Audit OAuthAudit `json:"audit"`
-}
-
-// OAuthAudit specifies the Audit profile in use.
-type OAuthAudit struct {
-	// profile is a simple drop in profile type that can be turned off by
-	// setting it to "None" or it can be turned on by setting it to
-	// "WriteLoginEvents".
-	// +kubebuilder:default:="WriteLoginEvents"
-	Profile OAuthAuditProfileType `json:"profile,omitempty"`
-}
-
-// OAuthAuditProfileType defines a simple audit profile, which can turn OAuth
-// authentication audit logging on or off.
-// +kubebuilder:validation:Enum=None;WriteLoginEvents
-type OAuthAuditProfileType string
-
-const (
-	// "None" disables audit logs.
-	OAuthNoneAuditProfileType AuditProfileType = "None"
-
-	// "WriteLoginEvents" logs login and login failure events.
-	// This is the default.
-	OAuthWriteLoginEventsProfileType AuditProfileType = "WriteLoginEvents"
-)
-```
+We propose to reuse the `audit.config.openshift.io/v1` profiles for enabling or disabling audit logging on the `oauth-server` and to add annotations to distinguish authentication events.
 
-We propose to add additional information that helps to identify login and login failures to the audit log.
+We propose that there is a limitation to log only the Metadata level as the content of the `oauth-server` is sensitive. The audit logging will be turned on, when the audit log policy profile is set to `Default`, `WriteRequestBodies` or `AllRequestBodies`. Audit logging will be turned of, when the audit log policy profile is set to `None` ([Configuring the audit policy](https://docs.openshift.com/container-platform/4.10/security/audit-log-policy-config.html)).
```

Comment:
```suggestion
We propose that there is a limitation to log only the Metadata level as the content of the `oauth-server` is considered security sensitive. The audit logging will be "just" turned on when the audit log policy profile is set to `Default`, `WriteRequestBodies` or `AllRequestBodies` and it will not adhere to the level of logging specified. Audit logging will be turned off, when the audit log policy profile is set to `None` ([Configuring the audit policy](https://docs.openshift.com/container-platform/4.10/security/audit-log-policy-config.html)).
```